content/arc-sqlmi-chaos-mesh/index.mdx

```diff
@@ -0,0 +1,248 @@
+
+- Living with them in Production for a while and accumulating lessons learned (or scars, if you asked a DBA); or,
+- Observing the system's behavior under _intense_ stress, the worst-imaginable failure conditions possible
+
+Previously, I had only on set up [Availability Groups or "AGs"](https://learn.microsoft.com/en-us/sql/database-engine/availability-groups/windows/overview-of-always-on-availability-groups-sql-server?view=sql-server-ver16) on Windows Server Failover Clusters (AGs are basically SQL Server's answer to resiliency guarantees in case of engine-level failure). Besides running [some one-off T-SQL commands](https://learn.microsoft.com/en-us/sql/database-engine/availability-groups/windows/perform-a-planned-manual-failover-of-an-availability-group-sql-server?view=sql-server-ver16) to manually failover/failback, there was always a "fear of the unknown" in my mind, regarding what would _actually_ happen, in case the underlying system and it's dependents completely fell over sideways. I also wasn't familiar with how to systematicaly orchestrate drastic failures in Windows, besides restarting a couple VMs or restarting `sqlservr.exe` from [`services.msc`](https://www.minitool.com/news/how-to-open-windows-services.html) etc.
+
+So recently, I was tasked with trying my absolute best _to break_ SQL Server Availability Group by any means possible. The definition of broken is:
+
+<Callout>
+
+"_Do whatever you want to the system (besides deleting `mdf/ldf` files etc., AKA no irrecoverable damage), for any duration of time, with the end goal of leaving the system in a state where it cannot self-heal, and requires manual intervention (e.g. intelligently crafted T-SQL commands that must be manually run by some human being)._
+
+_If it requires manual intervention, the system is broken, and that's a major bug._"
+
+</Callout>
+
+This time, I had Kubernetes on my side, with the SQL engine running in disposable Linux containers ([Azure Arc-enabled SQL Managed Instance, Business-Critical Tier](https://learn.microsoft.com/en-us/azure/azure-arc/data/service-tiers#service-tier-comparison)), and the latest-and-greatest Chaos Engineering software from CNCF - called [Chaos Mesh](https://www.cncf.io/projects/chaosmesh/) - offering many novel [Custom Resource Defintions](https://github.com/chaos-mesh/chaos-mesh#chaos-operator) that can be used to declare layers upon layers of controlled, systematic degradations (rather than one-off imperative scripts with a single entry and exit point like, restarting a VM, then moving on to cause some other mischief in a linear fashion, etc.).
```

Comment:
"point like, restarting a VM, then moving"

I would remove at least the first comma. The second is optional.

Reply:
Done